lib/tizen_build_target.dart

```diff
@@ -162,6 +166,60 @@ class TizenPlugins extends Target {
         sharedLib.copySync(outputDir.childFile(sharedLib.basename).path);
       }
     }
+
+    final Depfile pluginDepfile = collectPluginSources(
+      nativePlugins,
+      environment,
+      ephemeralDir,
+    );
+    final DepfileService depfileService = DepfileService(
+      fileSystem: environment.fileSystem,
+      logger: environment.logger,
+    );
+    depfileService.writeToFile(
+      pluginDepfile,
+      environment.buildDir.childFile('tizen_plugins.d'),
+    );
+  }
+
+  Depfile collectPluginSources(
+    List<TizenPlugin> nativePlugins,
+    Environment environment,
+    Directory ephemeralDir,
+  ) {
+    final List<File> inputs = <File>[];
+    final List<File> outputs = <File>[];
+
+    for (final TizenPlugin plugin in nativePlugins) {
+      for (final String arch in buildInfo.targetArchs) {
+        final TizenLibrary tizenLibrary = TizenLibrary(plugin.path);
+
+        final Directory headerDir = tizenLibrary.headerDir;
+        final Directory sourceDir = tizenLibrary.sourecDir;
+        final File projectFile = tizenLibrary.projectFile;
+
+        if (projectFile != null) {
+          inputs.add(projectFile);
+        }
+        if (headerDir?.existsSync() ?? false) {
+          headerDir.listSync().whereType<File>().forEach((File file) {
```

Comment:
Maybe you missed `recursive: true`?

Also you may prefer `forEach((File file) => inputs.add(file))`.